src/Election/Follower.java

```diff
@@ -1,21 +1,143 @@
 package Election;
 
+import java.util.Timer;
+
+import gash.router.server.ServerState;
+import pipe.common.Common;
+import pipe.election.Election.*;
+import pipe.election.Election.LeaderStatus.LeaderQuery;
+import pipe.election.Election.LeaderStatus.LeaderState;
 import pipe.work.Work.WorkMessage;
+import pipe.work.Work.WorkState;
+import io.netty.channel.Channel;
 
 public class Follower implements INodeState {
 
-	int lastHeartBeatValue;
-	ElectionTimer timer;
+	private long lastHeartBeatValue;
+	private Timer timer;
+	private ServerState state;
+	
 	
-	public Follower(){
-		//this.timer=;
+	public Follower(ServerState serverState){
+		this.state=serverState;
+		
+		timer.scheduleAtFixedRate(new ElectionTimer(this), 0, getDelay());
+	}
+
+	protected long getDelay() {
+		return state.getConf().getElectionTimeout();
 	}
 
 	@Override
-	public void HandleMessage(WorkMessage workMessage) {
-		// TODO Auto-generated method stub
+	public void handleMessage(WorkMessage workMessage, Channel channel) {
+		/*
+		 * Once I receive Heart beat message from leader, I update my lastHeartBeatValue.
+		 * */
+		if(workMessage.hasBeat())	{
+			setLastHeartbeat(System.currentTimeMillis());
+		}
 		
+		if(workMessage.getLeader().getAction() == LeaderQuery.VOTEREQUEST)	{
+			if(workMessage.getLeader().getElectionId() > state.getElectionId()){
+				VoteMessage vote = new VoteMessage(state.getConf().getNodeId(),workMessage.getLeader().getElectionId(),workMessage.getLeader().getLeaderId());
+				channel.writeAndFlush(vote.getMessage());
+			}
+			
+		}
+	}
+	
+	@Override
+	public void stateChanged()	{
+		//Start the timer
+		timer.scheduleAtFixedRate(new ElectionTimer(this), 0, getDelay());
+	}
+	
+	public long getLastHeartbeat() {
+		return lastHeartBeatValue;
+	}
+
+	public void setLastHeartbeat(long lastHeartbeat) {
+		this.lastHeartBeatValue = lastHeartbeat;
+	}
+	
+	public void isElectionNeeded(boolean result){
+		if(result)	{
+			timer.cancel();
+			state.setState(NodeStateEnum.CANDIDATE);	
+		}
 	}
 	
+	private class VoteMessage{
+		private WorkState.Builder workState;
+		private LeaderState leaderState;
+		private Common.Header.Builder header;
+		private LeaderStatus.Builder leaderStatus;
+		private int nodeId;
+		private int destination = -1; // By default Heart Beat Message will be sent to all Nodes..
```

Comment:
Did you mean vote message? In that case it should travel to expected destination.